Index: syslog-ng-3.6.2/configure.ac
===================================================================
--- syslog-ng-3.6.2.orig/configure.ac	2015-01-15 14:22:10.271436440 -0800
+++ syslog-ng-3.6.2/configure.ac	2015-01-15 14:28:12.093230618 -0800
@@ -116,6 +116,9 @@
 AC_ARG_ENABLE(memtrace,
               [  --enable-memtrace   Enable alternative leak debugging code.])
 
+AC_ARG_ENABLE(tls,
+              [  --enable-tls        Enable thread-local storage support.],,enable_tls="auto")
+
 AC_ARG_ENABLE(ssl,
               [  --enable-ssl        Enable SSL support.],,enable_ssl="auto")
 
@@ -407,12 +410,14 @@
 dnl Is the __thread keyword available?
 dnl ***************************************************************************
 
-AC_LINK_IFELSE([AC_LANG_PROGRAM(
-[[#include <pthread.h>
-__thread int a;
-]],
-[a=0;])],
-[ac_cv_have_tls=yes; AC_DEFINE_UNQUOTED(HAVE_THREAD_KEYWORD, 1, "Whether Transport Layer Security is supported by the system")])
+if test "x$enable_tls" != "xno"; then
+	AC_LINK_IFELSE([AC_LANG_PROGRAM(
+	[[#include <pthread.h>
+	__thread int a;
+	]],
+	[a=0;])],
+	[ac_cv_have_tls=yes; AC_DEFINE_UNQUOTED(HAVE_THREAD_KEYWORD, 1, "Whether Transport Layer Security is supported by the system")])
+fi
 
 dnl ***************************************************************************
 dnl How to do static linking?
