Index: tegra_multimedia_api/samples/v4l2cuda/Makefile
===================================================================
--- tegra_multimedia_api.orig/samples/v4l2cuda/Makefile
+++ tegra_multimedia_api/samples/v4l2cuda/Makefile
@@ -28,8 +28,6 @@ SRCS := \
 
 ALL_CPPFLAGS := $(addprefix -Xcompiler ,$(filter-out -std=c++11, $(CPPFLAGS)))
 
-GENCODE_FLAGS := $(NVCCFLAGS)
-
 all: $(APP)
 
 capture.o: capture.cpp
@@ -38,7 +36,7 @@ capture.o: capture.cpp
 
 yuv2rgb.o: yuv2rgb.cu
 	@echo "Compiling: $<"
-	$(NVCC) $(ALL_CPPFLAGS) $(GENCODE_FLAGS) -c $<
+	$(NVCC) $(ALL_CPPFLAGS) $(NVCCFLAGS) $(GENCODE_FLAGS) -c $<
 
 $(APP): capture.o yuv2rgb.o
 	@echo "Linking: $@"
